### PR TITLE
Issue #35: satelite projects should use latest checkstyle as soon at is appears in maven repo

### LINKED ISSUE #35: satelite projects should use latest checkstyle as soon at is appears in maven repo
Right now we do bump a version in a few projects https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release#update-for-released-version-reference right after release.

Let's find a way to define in pom.xml the latest snapshot and most recent version. And the latest released version when we define dependencies. To let us skip such updates at all.

CI of such projects should be able to get the latest version from https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml and run projects with such version.

We can have in all such repos, a GitHub action that runs once a day check https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml, put such values in required files, and send PRs with changes (PR is necessary as we need notice is any problems happens if we introduce breaking changes)

TASK:

- [ ] Version Bump in `ivy.xml`
- [ ] Version Bump in `pom.xml`
- [ ] Version Bump in `build.gradle`

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Partially Fixes https://github.com/sevntu-checkstyle/checkstyle-samples/issues/35

Testing:
```
rahul@rk7:~/Desktop/opensource/checkstyle-samples$ ./.ci/update-project-version.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3785  100  3785    0     0   1958      0  0:00:01  0:00:01 --:--:--  1960
LATEST_VERSION=10.3.2
CURRENT_VERSION=10.3
Checking Version for: ant-project/ivy.xml
Version updated to 10.3.2
rahul@rk7:~/Desktop/opensource/checkstyle-samples$ 
```

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
Screenshot from 2022-08-19 15-18-44

**Comment by Rahulkhinchi03** [comment]
The problem, I am facing: Extra Whitespace occurs at two other lines!

185594159-6e530949-2308-41e4-a7b8-d971024b3f2d

File: .ci/update-project-version.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+echo "Checking Version for: "$FILE
+if [ "$CURRENT_VERSION" != "$LATEST_VERSION"  ]; then
+  xmlstarlet edit --omit-decl -P --pf --inplace -u \
+    '//dependencies/dependency[@name="checkstyle"]/@rev' \
+    -v "$LATEST_VERSION" $FILE
+    echo "Version updated to $LATEST_VERSION"
+else
+echo "Skipped Update, Version is latest"
```
Comment (romani) [inline .ci/update-project-version.sh:22]: +2 spaces

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/update-project-version.sh (hunk 1 above)
Comment (Rahulkhinchi03) [reply .ci/update-project-version.sh]: Changes done in all other issues too!

File: .ci/bump-ant-project-version.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -e
```
Comment (romani) [inline .ci/bump-ant-project-version.sh:2]: apply comments from https://github.com/checkstyle/contribution/pull/635#pullrequestreview-1079599409

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by romani** [comment]
> I am facing: Extra Whitespace occurs at two other lines!

I am ok to merge extra spaces, if this is only extra stuff required, and next executions will not add more.

File: .ci/bump-ant-project-version.sh (hunk 2 above)
Comment (Rahulkhinchi03) [reply .ci/bump-ant-project-version.sh]: Changes Done

**Comment by romani** [comment]
I fixed file to be in good form.

@Rahulkhinchi03 , please review changes and keep doing like I did.

**Pull request merged**